packages/compiler/config/config-loader.ts

```diff
@@ -17,42 +18,53 @@ export const defaultConfig = deepFreeze({
 
 /**
  * Look for the project root by looking up until a `tspconfig.yaml` is found.
- * @param path Path to start looking
+ * @param path Path to the file or the folder to start looking
  */
 export async function findTypeSpecConfigPath(
   host: CompilerHost,
   path: string
 ): Promise<string | undefined> {
-  let current = path;
-  while (true) {
-    let pkgPath = await searchConfigFile(host, current, TypeSpecConfigFilename);
-    if (pkgPath === undefined) {
-      pkgPath = await searchConfigFile(host, current, OldTypeSpecConfigFilename);
-    }
-    // if found either file in current folder, return it
-    if (pkgPath !== undefined) {
-      return pkgPath;
-    }
-    const parent = getDirectoryPath(current);
-    if (parent === current) {
-      return undefined;
+  try {
+    const stats = fs.statSync(path);
+    if (stats.isFile()) {
+      return path;
+    } else if (stats.isDirectory()) {
+      let current = path;
+      while (true) {
+        let pkgPath = await searchConfigFile(host, current, TypeSpecConfigFilename);
+        if (pkgPath === undefined) {
+          pkgPath = await searchConfigFile(host, current, OldTypeSpecConfigFilename);
+        }
+        // if found either file in current folder, return it
+        if (pkgPath !== undefined) {
+          return pkgPath;
+        }
+        const parent = getDirectoryPath(current);
+        if (parent === current) {
+          return undefined;
+        }
+        current = parent;
+      }
     }
-    current = parent;
+    return undefined;
+  } catch (error) {
+    console.error(`ERROR: --config '${path}' not found. Using default configuration.`);
+    return undefined;
   }
 }
 
 /**
- * Load the typespec configuration for the provided directory
+ * Load the typespec configuration for the provided path or directory
  * @param host
- * @param directoryPath
+ * @param path
  */
 export async function loadTypeSpecConfigForPath(
   host: CompilerHost,
-  directoryPath: string
+  path: string
 ): Promise<TypeSpecConfig> {
-  const typespecConfigPath = await findTypeSpecConfigPath(host, directoryPath);
+  const typespecConfigPath = await findTypeSpecConfigPath(host, path);
   if (typespecConfigPath === undefined) {
-    return { ...deepClone(defaultConfig), projectRoot: directoryPath };
+    return { ...deepClone(defaultConfig), projectRoot: path };
```

Comment:
Does anything depend on projectRoot being a directory? It seems like a bad name if this is a file path now. Maybe we should get the directory of the specified path as the "root" and expose a separate thing to get the full config file path?